### ISSUE TITLE
Is it possible to create CLI that will let users `git clone` codesandboxes by URL

### ISSUE THREAD

**Issue by nikitavoloboev** [codesandbox/codesandbox-client#3011]
I want to effectively be able to run 

`git clone https://codesandbox.io/s/dry-microservice-qnle3` 

And have the codesandbox be cloned as project of files in my file system.

I was thinking of creating a tool like [hub](https://github.com/github/hub) that lets you do it.

Is there CodeSandbox API I can use to create such tool. Or perhaps such functionality already exists?

My use case is I often see codesandbox I like but I still prefer my environment of VS Code than browser so I want to clone it and open in my main editor.

Thank you.

**Comment by lbogdan** [comment]
Hey @nikitavoloboev ,

We've been thinking about exposing sandboxes as git repositories for a while, and it's on our road-map, but we don't have an ETA yet. Until then, you can download sandboxes using File -> Export to ZIP. Does this address your needs?

**Comment by nikitavoloboev** [comment]
> Does this address your needs?

It doesn't in that I want to automate it in similar way as described here:

https://medium.com/@nikitavoloboev/insta-cloning-ff5f38eb1d32

Clicking and then exporting it to ZIP, and then having to open up the ZIP is not great.

**Comment by reed-jones** [comment]
Even an endpoint to download the zip via wget or curl would be great!

**Comment by SaraVieira** [comment]
Hey

The reason this doesn't exist for now is that we package them and zip them in the frontend on demand so thats why the api doesn't provide a URL even though it is possible but needs to be done in our end :(

**Comment by lbogdan** [comment]
Hey @nikitavoloboev ,

Are you still interested in contributing this? I just realized that I completely overlook your question about an existing API to help you with this. We have in fact an API we're calling from the UI to get a sandbox's files, the endpoint is `https://codesandbox.io/api/v1/sandboxes/:sandboxId`, e.g. for this sandbox https://codesandbox.io/s/vue-ministore-foumd the "get sandbox" API URL is https://codesandbox.io/api/v1/sandboxes/foumd .

Please let me know if you need futher assistance with this!

**Comment by nikitavoloboev** [comment]
I will look into it in a bit. Will get back to you.

**Comment by SaraVieira** [comment]
Hey

Did you take a look into this?

Is this something you would like to pursue?

**Comment by nikitavoloboev** [comment]
> Did you take a look into this?

I have not. I saw a trick somewhere where given a GitHub url, you can change something and it will open the github repo in CodeSandbox.

**Comment by SaraVieira** [comment]
Is it this?

https://github.com/dferber90/githubbox

**Comment by nikitavoloboev** [comment]
Yes, I think so. Just tried it on codesandbox with https://githubbox.com/codesandbox/codesandbox-client/issues/3011 but it seems to be loading it in forever

<img width="1232" alt="image" src="https://user-images.githubusercontent.com/6391776/92124624-4bf45000-edfe-11ea-987a-44add46fa6f1.png">

Perhaps because CodeSandbox is a large repo. Not sure as there is no progress indicator or anything.

**Comment by nikitavoloboev** [comment]
Ok it finished with error

<img width="790" alt="image" src="https://user-images.githubusercontent.com/6391776/92124743-747c4a00-edfe-11ea-93a3-596a21206e55.png">

**Comment by SaraVieira** [comment]
Hey

yeah that will never open, we have a max of files ahah
